### PR TITLE
Fix setuptools regression

### LINKED ISSUE #133: [Bug]: Action "Project Creation" fails on main branch
### What happened?

A regression on main, it worked about a week ago

### Operating System

Linux

### What release version or branch did you use?

main

### Relevant log output

```shell
2024-07-04T08:37:28.7486711Z app/tests/integration/integration_test.py:43: AssertionError
2024-07-04T08:37:28.7487984Z ------------------------------ Captured log call -------------------------------
2024-07-04T08:37:28.7489114Z ERROR    grpc._common:_common.py:91 Exception serializing message!
2024-07-04T08:37:28.7490001Z Traceback (most recent call last):
2024-07-04T08:37:28.7491312Z   File "/home/vscode/.local/lib/python3.10/site-packages/grpc/_common.py", line 89, in _transform
2024-07-04T08:37:28.7492410Z     return transformer(message)
2024-07-04T08:37:28.7493947Z TypeError: descriptor 'SerializeToString' for 'google._upb._message.Message' objects doesn't apply to a 'list' object
2024-07-04T08:37:28.7495395Z =========================== short test summary info ============================
2024-07-04T08:37:28.7496585Z FAILED app/tests/integration/integration_test.py::test_set_position_not_allowed
2024-07-04T08:37:28.7497740Z !!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!
2024-07-04T08:37:28.7498676Z ============================== 1 failed in 21.11s ==============================
```
```


### Code of Conduct

- [X] I agree to follow this project's rules for contributions

### PR DESCRIPTION
**Pull request by erikbosch**
This fixes the build problem by upgrading pyinstaller. It also addresses various build warnings by upgrading actions and refactoring the license check similar to what have been in some other repos.

See also https://github.com/pypa/setuptools/issues/4374

Fixes #133

### CODE REVIEW

File: .github/workflows/project-creation.yaml [hunk 1]
```diff
@@ -27,6 +27,9 @@ on:
   pull_request:
     branches:
       - main
+  # Scheduled build to fetch regressions, if any
```
Comment (erikbosch) [inline .github/workflows/project-creation.yaml:30]: Good to test daily if some regressions have occured

**Comment by erikbosch** [comment]
@lukasmittag - this seemed to do the trick. I have only updated the example causing the regression though

File: examples/seat-adjuster/Dockerfile [hunk 2]
```diff
@@ -44,7 +44,7 @@ RUN velocitas init -v
 
 WORKDIR /workspace/app
 
-RUN pyinstaller --clean -F -s -p src src/main.py
+RUN pyinstaller --hiddenimport=pkg_resources.extern --clean -F -s -p src src/main.py
```
Comment (erikbosch) [inline examples/seat-adjuster/Dockerfile:47]: I will also try to remove the fixed `pyinstaller` version above, introuced in https://github.com/eclipse-velocitas/vehicle-app-python-sdk/pull/85

The  `pyinstaller` shall have been fixed (see https://github.com/pyinstaller/pyinstaller/issues/8554). I will test if just removing the fixed version will solve the problem or if we manually must fetch a newer pyinstaller version.

File: .project-creation/.skeleton/Dockerfile [hunk 3]
```diff
@@ -32,7 +32,7 @@ RUN /bin/bash -c 'set -ex && \
     pip3 install --no-cache-dir scons; \
     fi'
 
-RUN pip3 install --no-cache-dir pyinstaller==5.9.0 \
+RUN pip3 install --no-cache-dir pyinstaller \
```
[the remark this answers is not in the document]
Comment (BjoernAtBosch) [reply .project-creation/.skeleton/Dockerfile:35]: @erikbosch: There is a comment in the old PR #85: I think, I just did it as it is/was (?) also pinned in the vehicle_app_python_template repo

[the remark this answers is not in the document]
Comment (BjoernAtBosch) [reply .project-creation/.skeleton/Dockerfile:35]: I guess, the reason for pinning the pyinstaller is, that it creates the package of the app and that behaviour shall be stable, i.e. o guaranteed to be reproducibility. The danger that a new version is changing something important might be low in this case, but I'm not sure. Maybe, we should pin the (new) major version at least?

[the remark this answers is not in the document]
Comment (erikbosch) [reply .project-creation/.skeleton/Dockerfile:35]: If ww want to pin - which version do we want to pin against. Current version is 6.9.0, they rarely release patches, so shall we use `==6.9` just incase?

**Comment by erikbosch** [comment]
PR updated now - now no warnings on builds given. If we go for pyinstaller 6.9 as in this PR we could update other builds afterwards

**Comment by erikbosch** [comment]
@BjoernAtBosch @lukasmittag - updated now. Going for 6.9 as pinned version?

**Review by BjoernAtBosch — APPROVED** [review]
LGTM

**Pull request merged**